aristanetworks/ansible-cvp
# Add ansible check mode support

## Types of changes


## Related Issue(s)

Fix #134 

## Proposed changes

Add code in modules to guess changes from CVP side to support dry-run mode.

A check is done on `module.check_mode` to see if we can send API call to CV or just provides fake result of the change.

## How to test

```yaml
---
- name: cv_configlet unit testing
  hosts: cv_server
  connection: local
  gather_facts: false
  collections:
    - arista.avd
    - arista.cvp
  vars:
    CVP_CONFIGLETS:
      01TRAINING-alias: "alias a101 show version"
      01TRAINING-01: "alias a102 show version"
  tasks:
    - name: "Gather CVP facts {{inventory_hostname}}"
      arista.cvp.cv_facts:
        facts:
          configlets
      register: CVP_FACTS

    - name: "Configure configlet on {{inventory_hostname}}"
      arista.cvp.cv_configlet:
        cvp_facts: "{{CVP_FACTS.ansible_facts}}"
        configlets: "{{CVP_CONFIGLETS}}"
        configlet_filter: ["TRAINING"]
        state: present
      register: CVP_CONFIGLET_RESULT

    - name: "Print logs"
      debug:
        msg: "{{ CVP_CONFIGLET_RESULT }}"
```

And then run 

```
$ ansible-playbook playbooks/unit-cv-configlets.yml --extra-vars "configlets_prefix_var=ASE" --check
```

## Checklist:


- [x] I have read the [**CONTRIBUTING**](https://github.com/aristanetworks/ansible-cvp/blob/master/contributing.md#branches) document.
- [ ] All new and existing tests passed (`make linting` and `make sanity-lint`).
